iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_queue_popper.inl

```diff
@@ -119,28 +119,22 @@ inline void ChunkQueuePopper<ChunkQueueDataType>::clear() noexcept
 }
 
 template <typename ChunkQueueDataType>
-inline void ChunkQueuePopper<ChunkQueueDataType>::setConditionVariable(
-    cxx::not_null<ConditionVariableData*> conditionVariableDataPtr) noexcept
+inline void ChunkQueuePopper<ChunkQueueDataType>::setConditionVariable(ConditionVariableData& conditionVariableDataRef,
+                                                                       const uint64_t notificationIndex) noexcept
 {
     typename MemberType_t::LockGuard_t lock(*getMembers());
 
-    getMembers()->m_conditionVariableDataPtr = conditionVariableDataPtr;
-}
-
-template <typename ChunkQueueDataType>
-inline void ChunkQueuePopper<ChunkQueueDataType>::setEventVariable(EventVariableData& eventVariableDataPtr,
-                                                                   const uint64_t eventId) noexcept
-{
-    getMembers()->m_conditionVariableDataPtr = &eventVariableDataPtr;
-    getMembers()->m_eventVariableIndex.emplace(eventId);
+    getMembers()->m_conditionVariableDataPtr = &conditionVariableDataRef;
+    getMembers()->m_conditionVariableNotificationIndex = notificationIndex;
 }
 
 template <typename ChunkQueueDataType>
 inline void ChunkQueuePopper<ChunkQueueDataType>::unsetConditionVariable() noexcept
 {
     typename MemberType_t::LockGuard_t lock(*getMembers());
+
     getMembers()->m_conditionVariableDataPtr = nullptr;
-    getMembers()->m_eventVariableIndex.reset();
+    getMembers()->m_conditionVariableNotificationIndex = ConditionNotifier::INVALID_NOTIFICATION_INDEX;
```

Comment:
I would keep the surrogate value, but change it into an optional for the chunk queue. The ConditionNotifier should have always an index which is valid but I feel somehow safer when we set `m_notificationIndex` to INVALID_NOTIFICATION_INDEX in the declaration. It can be less error prone when we someday later introduce a default ctor or adding moving/copying capabilities.